rosetta/transactions/intent.go

```diff
@@ -101,17 +88,19 @@ func (p *Parser) CreateTransactionIntent(operations []object.Operation) (*Intent
 
 	// Make sure that both the send and receive operations use the same currency.
 	// This is perhaps unnecessary at the moment since we only have a single currency.
-	if send.Amount.Currency != receive.Amount.Currency {
+	if send.Amount.Currency.Symbol != receive.Amount.Currency.Symbol {
 		return nil, failure.InvalidIntent{
-			Sender:      send.AccountID.Address,
-			Receiver:    receive.AccountID.Address,
-			Description: failure.NewDescription("send and receive currencies do not match"),
+			Sender:   send.AccountID.Address,
+			Receiver: receive.AccountID.Address,
+			Description: failure.NewDescription("send and receive currencies do not match",
+				failure.WithString("withdrawal_currency", send.Amount.Currency.Symbol),
+				failure.WithString("deposit_currency", receive.Amount.Currency.Symbol)),
 		}
 	}
```

Comment:
I still don't think we should allow any symbol here but `dps.FlowSymbol`, so I would just hard-code a check on both symbols for `FLOW`.